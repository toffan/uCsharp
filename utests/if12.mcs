int main() {
    int c1 = 3;
    int c2 = 0;

    if (c1) {
        if (!c2) {
            int b = 2;
        } else {
            int a = 3;
        }
        int a = 5;
    } else {
        int b = 1;
    }

    return 0;
}
