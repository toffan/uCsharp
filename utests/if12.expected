JUMP __main
_main:
    LOADL 3
    LOADL 0
    LOAD (1) 3[LB]
    JUMPIF (0) _X0:
    LOAD (1) 4[LB]
    JUMPIF (0) _X2:
    LOADL 2
    JUMP _X3:
    _X2:
    LOADL 3
    _X3:
    LOADL 5
    JUMP _X1:
    _X0:
    LOADL 1
    _X1:

    LOADL 0
    RETURN (1) 0
    RETURN (0) 0
__main:
    CALL (SB) _main
    HALT
